consensus/src/processes/pruning_proof/validate.rs

```diff
@@ -145,23 +144,24 @@ impl PruningProofManager {
                 continue;
             }
 
-            match relations_read[level_idx].get_parents(current_pp).unwrap_option() {
-                Some(parents) => {
-                    if parents.iter().copied().any(|parent| {
-                        current_consensus_ghostdag_stores[level_idx].get_blue_score(parent).unwrap() < 2 * self.pruning_proof_m
-                    }) {
-                        return Ok(());
-                    }
-                }
-                None => {
-                    // If the current pruning point doesn't have a parent at this level, we consider the proof state to be better.
-                    return Ok(());
-                }
+            let parents_at_level: Vec<_> = self
+                .parents_manager
+                .parents_at_level(&current_pp_header, level)
+                .iter()
+                .copied()
+                .filter_map(|parent| current_consensus_ghostdag_stores[level_idx].get_blue_score(parent).unwrap_option())
+                .collect();
+            if parents_at_level.iter().copied().any(|parent_bscore| parent_bscore < 2 * self.pruning_proof_m) {
+                return Ok(());
+            }
+
+            if parents_at_level.is_empty() {
+                // If the current pruning point doesn't have a parent at this level, we consider the proof state to be better.
+                return Ok(());
```

Comment:
```suggestion
            let current_consensus_max_blue_score = self
                .parents_manager
                .parents_at_level(&current_pp_header, level)
                .iter()
                .copied()
                .filter_map(|parent| current_consensus_ghostdag_stores[level_idx].get_blue_score(parent).unwrap_option())
                .max()
                .unwrap_or_default();
            
            if current_consensus_max_blue_score < 2 * self.pruning_proof_m {
                return Ok(());
```

The original logic should have been `.all(..)` rather than `.any(..)`, because we should prefer current consensus even if only one of pp parents at level has `blue score >= 2M`. 

Given that (cc @someone235), the above eqv suggested code (with `max` replacing the `all`) is a cleaner and clearer implementation